src/edit/index.vue

```diff
@@ -1,9 +1,138 @@
 <template>
-  <p>Custom element template</p>
+  <div class="tce-api-video">
+    <element-placeholder
+      v-if="showPlaceholder"
+      :is-focused="isFocused"
+      :is-disabled="isDisabled"
+      name="Api Video component"
+      icon="mdi-video-image"
+      active-placeholder="Use toolbar to upload the video"
+      active-icon="mdi-arrow-up" />
+    <div v-else>
+      <div v-if="showError" class="overlay">
+        <div class="message error--text">
+          <v-icon class="error--text">mdi-alert</v-icon>
+          {{ error || 'Error loading media!' }}
+        </div>
+      </div>
+      <div v-if="!showError && infoMessage" class="overlay">
+        <div class="message info--text">
+          <v-progress-circular indeterminate color="info" class="mr-4" />
+          {{ infoMessage }}
+        </div>
+      </div>
+      <div ref="player" class="player"></div>
+    </div>
+  </div>
 </template>
 
 <script>
+import { ELEMENT_STATE } from '../shared';
+import ElementPlaceholder from '../tce-core/ElementPlaceholder.vue';
+import get from 'lodash/get';
+import upload from '../createUpload';
+
+const DEFAULT_ERROR_MSG = 'Something went wrong.';
+const CANCEL_UPLOAD_ERROR_MSG = 'Upload canceled by leaving the page.';
+const UPLOADING_MSG = 'Video is uploading... Do not leave the page.';
+const PROCESSING_MSG = 'Video is processing...';
+
 export default {
-  name: 'tce-api-video'
+  name: 'tce-api-video',
+  inject: ['$elementBus'],
+  props: {
+    element: { type: Object, required: true },
+    isFocused: { type: Boolean, default: false },
+    isDisabled: { type: Boolean, default: false }
+  },
+  data: () => ({ file: null }),
+  computed: {
+    videoId: ({ element }) => element.data?.videoId,
+    uploadUrl: ({ element }) => element.data?.uploadUrl,
+    embedCode: ({ element }) => element.data?.embedCode,
+    fileName: ({ element }) => element.data?.fileName,
+    playable: ({ element }) => element.data?.playable,
+    status: ({ element }) => element.data?.status,
+    error: ({ element }) => element.data?.error,
+    showError: ({ status }) => status === ELEMENT_STATE.ERROR,
+    showPlaceholder: ({ error, fileName }) => !error && !fileName,
```

Comment:
A name like `isEmptyOrError` might make it easier to infer from the template what's rendered and when.

Reply:
I think `isEmptyOrError` is not describing this case well. Here we have a case where we need to check if it is empty and there is **no** error. 
`isFileOrError` better fits this use-case, but personally, I like `showPlaceholder` more. What do you think?

Reply:
My bad, I misread it. The idea was to find the name that'd describe _why_ is the placeholder shown, so it makes it easy to figure out the logic from reading the template.
Maybe isPristine/isEmpty would suffice in that matter?

It's a minor detail. If you think it makes sense we can keep it the way it is. 😁